engine/table/src/main/java/io/deephaven/engine/table/impl/BlinkTableTools.java

```diff
@@ -130,62 +117,49 @@ private static Table internalBlinkToAppendOnly(final Table blinkTable, boolean h
-                                    final long totalSize = currentSize + newRowsSize;
-                                    columns.values().forEach(c -> c.ensureCapacity(totalSize));
-                                    final RowSet newRange = RowSetFactory.fromRange(currentSize, totalSize - 1);
-
-                                    ChunkUtils.copyData(sourceColumns, newRowSet, destColumns, newRange, false);
-                                    rowSet.insertRange(currentSize, totalSize - 1);
-
-                                    final TableUpdateImpl downstream = new TableUpdateImpl();
-                                    downstream.added = newRange;
-                                    downstream.modified = RowSetFactory.empty();
-                                    downstream.removed = RowSetFactory.empty();
-                                    downstream.modifiedColumnSet = ModifiedColumnSet.EMPTY;
-                                    downstream.shifted = RowSetShiftData.EMPTY;
-                                    result.notifyListeners(downstream);
-
-                                    // TODO Should this be left here
-                                    if (hasMaxRowLimit && totalSize > maxRowLimit) {
-                                        throw new IllegalStateException(
-                                                "Size of table cannot exceed the allowable limit, size=" + totalSize
-                                                        + ", limit=" + maxRowLimit);
-                                    }
-
-                                    if (hasMaxRowLimit && totalSize == maxRowLimit) {
-                                        // No more rows can be appended
-                                        // TODO Check if this needs to be kept here
-                                        blinkTable.removeUpdateListener(this);
-                                        return;
-                                    }
+                        swapListener.setListenerAndResult(new BaseTable.ListenerImpl("streamToAppendOnly",
+                                baseBlinkTable, result) {
+                            @Override
+                            public void onUpdate(TableUpdate upstream) {
+                                if (upstream.modified().isNonempty() || upstream.shifted().nonempty()) {
+                                    throw new IllegalArgumentException("Blink tables should not modify or shift!");
                                 }
-                            }, result);
-                        }
+                                RowSet newRowSet = upstream.added();
+                                long newRowsSize = newRowSet.size();
+                                if (newRowsSize == 0) {
+                                    return;
+                                }
+                                final long currentSize = rowSet.size();
+                                if (currentSize + newRowsSize >= maxRowLimit) {
```

Comment:
I was going to suggest some code to protect against overflowing a Long but that seems pretty silly given the size of a table that would overflow.  So ignore this comment.